examples/c/tcp-push-pop.c

```diff
@@ -51,18 +58,19 @@ static int pop_get_received_nbytes(int sockqd)
     return recv_bytes;
 }
 
-static void run_server(struct sockaddr_in *addr)
+static void run_server(struct sockaddr_in *addr, size_t data_size, unsigned max_msgs)
 {
-    unsigned int recv_bytes = 0;
+    size_t recv_bytes = 0;
     int sockqd = accept_get_newsockqd(addr);
+    size_t max_bytes = data_size * max_msgs;
 
-    while (recv_bytes < MAX_BYTES)
+    while (recv_bytes < max_bytes)
     {
         recv_bytes += pop_get_received_nbytes(sockqd);
-        fprintf(stdout, "pop: total bytes received: (%d)\n", recv_bytes);
+        fprintf(stdout, "pop: total bytes received: (%zu)\n", recv_bytes);
     }
 
-    assert(recv_bytes == MAX_BYTES);
+    assert(recv_bytes == max_bytes);
```

Comment:
This assert seems redundant since it seems impossible to exit the while loop without recv_bytes == max_bytes. Could be removed.

Reply:
Indeed.